openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIParameterMapper.java

```diff
@@ -79,6 +90,17 @@ public void getResourceInputs(Components components, SemanticModel semanticModel
             if (parameterNode.kind() == SyntaxKind.REQUIRED_PARAM) {
                 RequiredParameterNode requiredParameterNode = (RequiredParameterNode) parameterNode;
                 // Handle query parameter
+                if (requiredParameterNode.typeName().kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                    QualifiedNameReferenceNode referenceNode =
+                            (QualifiedNameReferenceNode) requiredParameterNode.typeName();
+                    String typeName = (referenceNode).modulePrefix().text() + ":" + (referenceNode).identifier().text();
+                    if (typeName.equals("http:Request")) {
+                        DiagnosticMessages messages = DiagnosticMessages.OAS_CONVERTOR_104;
+                        IncompatibleResourceDiagnostic error = new IncompatibleResourceDiagnostic(messages.getCode(),
```

Comment:
Since we are using all the fields of `DiagnosticMessages`, we could improve the constructor of `IncompatibleResourceDiagnostic` as follows:
```java
    public IncompatibleResourceDiagnostic(DiagnosticMessages message, Location location) {
        this.code = message.getCode();
        this.message = messages.getDescription();
        this.location = location;
        this.severity = messages.getSeverity();
    }
```

Reply:
There are scenarios, we won't be able to take a complete message from  `DiagnosticMessages`. Since we need to modify that error message by adding the relevant field name when we are going to create `IncompatibleResourceDiagnostic `.  Shall we keep the message field as parameter there. WDYT? 🙂

Reply:
I think we could improve the above constructor signature further more to support your requirement.
```java
public IncompatibleResourceDiagnostic(DiagnosticMessages message, Location location, String... args) {
  // implement logic here
}
```

Reply:
Sure I will make two constructors, make one with args  and add a string template for the enum 🙂